src/pudl_rmi/modernize.py

```diff
@@ -0,0 +1,149 @@
+"""
+WIP!!! Modernize the depreciation studies.
+
+This is UNDER CONSTRUCTION. Right now, this only supports generating the
+modernizable records that we've been using to mock up the modernization process
+for Duke.
+"""
+
+import logging
+
+import numpy as np
+import pandas as pd
+
+import pudl_rmi
+
+logger = logging.getLogger(__name__)
+
+#################################
+# Temp functions for Duke outputs
+#################################
+
+
+def fake_duke_deprish_eia_for_mod(df_to_fake, ppe):
+    """Temp function to fake Duke's deprish records for modernization."""
+    logger.info("Adding fake years of Duke data....")
+    # og_index = df_to_fake.index.names
+    # df_to_fake = df_to_fake.reset_index()
+    fake_year_dfs = []
+    to_fake_2018 = df_to_fake[
+        df_to_fake.utility_id_pudl.isin([90, 97])
+        & (df_to_fake.report_date.dt.year == 2018)
+    ].reset_index()[
+        [
+            c
+            for c in df_to_fake.reset_index()
+            if c
+            in ["record_id_eia", "line_id"]  # core IDs
+            + pudl_rmi.deprish.IDX_COLS_DEPRISH  # deprish IDS
+        ]
+    ]
+    for fake_year in [2019, 2020]:
+        fake_new_year = (
+            to_fake_2018.copy()
+            .assign(
+                report_year=fake_year,
+                report_date=pd.to_datetime(fake_year, format="%Y"),
+            )
+            .replace(
+                {
+                    "record_id_eia": "_2018_",
+                    "line_id": "2018_",
+                },
+                {"record_id_eia": f"_{fake_year}_", "line_id": f"{fake_year}_"},
+                regex=True,
```

Comment:
Why are there two separate dictionaries here? This is more for my own learning...

Reply:
Inline is always nice for method chaining, so I hear ya